examples/example_optimize_rrt_path.py

```diff
@@ -0,0 +1,136 @@
+    CubicTrajectoryOptimizationOptions,
+)
+from pyroboplan.visualization.meshcat_utils import visualize_frames
+
+
+if __name__ == "__main__":
+    # Create models and data.
+    model, collision_model, visual_model = load_models()
+    add_self_collisions(model, collision_model)
+    add_object_collisions(model, collision_model, visual_model)
+    data = model.createData()
+
+    # Initialize visualizer.
+    viz = MeshcatVisualizer(model, collision_model, visual_model, data=data)
+    viz.initViewer(open=True)
+    viz.loadViewerModel()
+
+    def random_valid_state():
+        return get_random_collision_free_state(
+            model, collision_model, distance_padding=0.01
+        )
+
+    while True:
+        q_start = random_valid_state()
+        q_goal = random_valid_state()
+        viz.viewer["rrt_start"].delete()
+        viz.viewer["rrt_goal"].delete()
+        viz.viewer["planned_path"].delete()
+        viz.viewer["optimized_trajectory"].delete()
+        viz.display(q_start)
+        time.sleep(1.0)
+
+        # Search for a path
+        options = RRTPlannerOptions(
+            max_step_size=0.05,
+            max_connection_dist=0.25,
+            rrt_connect=False,
+            bidirectional_rrt=True,
+            rrt_star=True,
+            max_rewire_dist=3.0,
+            max_planning_time=10.0,
+            fast_return=True,
+            goal_biasing_probability=0.15,
+            collision_distance_padding=0.01,
+        )
+        print("")
+        print(f"Planning a path...")
+        planner = RRTPlanner(model, collision_model, options=options)
+        q_path = planner.plan(q_start, q_goal)
+        if len(q_path) > 0:
+            print(f"Got a path with {len(q_path)} waypoints")
+            planner.visualize(viz, "panda_hand", show_tree=True)
+        else:
+            print("Failed to plan.")
+            continue
+
+        # Perform trajectory optimization.
+        dt = 0.025
+        options = CubicTrajectoryOptimizationOptions(
+            max_planning_time=30.0,
+            num_waypoints=len(q_path),
+            samples_per_segment=11,
+            min_segment_time=0.5,
+            max_segment_time=10.0,
+            max_total_time=12.0,
+            min_vel=-1.5,
+            max_vel=1.5,
+            min_accel=-0.75,
+            max_accel=0.75,
+            min_jerk=-1.0,
+            max_jerk=1.0,
+            check_collisions=True,
+            min_collision_dist=0.001,
+            collision_influence_dist=0.05,
+            collision_link_list=[
+                "obstacle_box_1",
+                "obstacle_box_2",
+                "obstacle_sphere_1",
+                "obstacle_sphere_2",
+                "ground_plane",
+            ],
+        )
+        print("Optimizing the path...")
+        optimizer = CubicTrajectoryOptimization(model, collision_model, options)
+        traj = optimizer.plan([q_path[0], q_path[-1]], init_path=q_path)
+
+        if traj is not None:
```

Comment:
Maybe add something like

```

        else:
            print("Trajectory optimization failed")
            input("Press 'Enter' to try again")
```
To give users a chance to look at the path that failed optimization. It's nifty to see what fails/succeeds now.